zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -1553,4 +1554,21 @@ private <T> boolean checkPermission(String noteId,
       return false;
     }
   }
+
+  private String decodeRepeatedly(final String encoded) throws IOException {
+    String previous = encoded;
+    int maxDecodeAttempts = 5;
+    int attempts = 0;
+
+    while (attempts < maxDecodeAttempts) {
+      String decoded = URLDecoder.decode(previous, StandardCharsets.UTF_8.toString());
```

Comment:
```suggestion
      String decoded = URLDecoder.decode(previous, StandardCharsets.UTF_8);
```
After https://github.com/apache/zeppelin/commit/414f4d1c470b5046628cfc378e6602fb78240e8f, we can now use JDK 11 functions.